### ISSUE TITLE
sound: finish/restructure sound modules

### ISSUE THREAD

**Issue by jiaoboxuan** [uchicago-cs/chiventure#1319]
Summary: Currently trying to write out the `sound.c` file. I presume that for actions that require sound (action sound effects, bgm, game state, etc), `sound.h` will have to be included in the respective c file. The process for writing the functions found in the sound.h file (load, free, and play) can be found in demo.c in the sandbox file. We hope to generalize this process to create the functions. Hopefully these functions in the sound.c file will be tested in the future and we hope the functions will be able to be used by other teams to be integrated into chiventure so we will also need to make documentation.

Closing statement: `sound.c` is easier than expected; it's basically initialization, playing sound, and freeing `sound_t`. Through the process I found out the difference between  `Mix_Chunk` and `Mix_Music` arrays. The former could play multiple sounds simulataneously (as long as they are loaded, of course), while the latter could play only one sound at at a time (which would be preferable with background music). Right now, we are having issues with the music channels, likely due to a library dependency error. The wav channels are doing well.

**Comment by JaStarbuck** [comment]
Created functions, pushed to the branch sound/background_music. Running into the error of the include statement:
#include "common/uthash.h"
Will need to figure out this error.

**Comment by JaStarbuck** [comment]
The functions written were based off the code, more generalized and split apart though, from demo.c.

**Comment by JaStarbuck** [comment]
Putting this comment from above here:
Closing statement: sound.c is easier than expected; it's basically initialization, playing sound, and freeing sound_t. Through the process I found out the difference between Mix_Chunk and Mix_Music arrays. The former could play multiple sounds simulataneously (as long as they are loaded, of course), while the latter could play only one sound at at a time (which would be preferable with background music). Right now, we are having issues with the music channels, likely due to a library dependency error. The wav channels are doing well.
